File: Assignment2/program5.c
////////////////////////////////////////////////////////////////
//
//  File Name   : program5.c
//  Description : Checks the entered Number is Even or Odd
//  Author      : Akhilesh.P.Sonavane
//  Date        : 10/05/2025
//
////////////////////////////////////////////////////////////////

#include<stdio.h>

typedef int BOOL;
#define TRUE 1
#define FALSE 0

////////////////////////////////////////////////////////////////
// 
//  Function Name : ChkEven
//  Description   : Checks the input Integer is Even or Odd 
//  Input         : Integer
//  Output        : Boolean
//
////////////////////////////////////////////////////////////////

BOOL ChkEven(int iNo)
{
    if((iNo % 2) == 0)
    {
        return TRUE;
    }
    else
    {
        return FALSE;
    }
}

int main()
{
    int iValue = 0;
    BOOL bRet = FALSE;

    printf("Enter Number\n");
    scanf("%d",&iValue);

    bRet = ChkEven(iValue);
    
    if(bRet == 1)
    {
        printf("The Entered Integer %d is Even Number\n",iValue);
    }
    else 
    {
        printf("The Entered Integer %d is Odd Number\n",iValue);
    }
    
    return 0;
}

File: Assignment17/program5.java
////////////////////////////////////////////////////////////////
//
//  File Name   : program5.java
//  Description : Accept a Integer and Print 'Diagonal Integer' Pattern
//  Author      : Akhilesh.P.Sonavane.
//  Date        : 22/05/2025
//
////////////////////////////////////////////////////////////////

import java.util.*;

////////////////////////////////////////////////////////////////
// 
//  Class Name    : Pattern
//  Function Name : Display
//  Description   : Inputs a Integer and returns 'Diagonal Integer' Pattern
//  Input         : Integer
//  Output        : Integer
//
////////////////////////////////////////////////////////////////

class Pattern 
{
    public void Display(int iRow, int iCol)
    {
        int i = 0, j = 0;
        
        if(iRow != iCol)
        {
            System.out.println("Invalid Input");
            return;
        }

        for(i = 1; i <= iRow; i++)
        {
            for(j = 1; j <= iCol; j++)
            {
                if((i == 1 || i == iRow) || (j == 1 || j == iCol) || (i == j))
                {
                    System.out.print(j+"\t");
                }
                else 
                {
                    System.out.print("\t");
                }
            }
            System.out.print("\n\n");
        }
    }
}
public class program5
{
    public static void main(String[] args) 
    {
        int iValue1 = 0;
        int iValue2 = 0;

        Scanner sobj = new Scanner(System.in);
        
        System.out.println("Enter Number of Rows : ");
        iValue1 = sobj.nextInt();

        System.out.println("Enter Number of Coloumns : ");
        iValue2 = sobj.nextInt();

        Pattern pobj = new Pattern();

        pobj.Display(iValue1,iValue2);

        sobj.close();
    }
}
